src/traces/ohlc/style.js

```diff
@@ -0,0 +1,35 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var d3 = require('d3');
+var Drawing = require('../../components/drawing');
+var Color = require('../../components/color');
+
+module.exports = function style(gd, cd) {
+    var s = cd ? cd[0].node3 : d3.select(gd).selectAll('g.ohlclayer').selectAll('g.trace');
+
+    s.style('opacity', function(d) {
+        return d[0].trace.opacity;
+    });
+
+    s.each(function(d) {
+        var trace = d[0].trace;
+
+        d3.select(this).selectAll('path').each(function(di) {
+            var dirLine = trace[di.dir].line;
+            d3.select(this)
+                .style('fill', 'none')
+                .call(Color.stroke, dirLine.color)
+                .call(Drawing.dashLine, dirLine.dash, dirLine.width)
```

Comment:
In the `finance_style` mock, the dash line pattern makes the new split increasing/decreasing legend item look a little strange:

![image](https://user-images.githubusercontent.com/6675409/38830975-f5737d12-418b-11e8-98ab-9b83ceb1356c.png)

compare to e.g. for the `ohlc_first` mock with `layout.showlegend: true`:

![image](https://user-images.githubusercontent.com/6675409/38831043-35d3d03c-418c-11e8-8e83-2e9f9255fab3.png)

which looks amazing.

Should we just not call `Drawing.dashLine` here?

Reply:
True, it looks weird... but it also looks weird in the plot. I think that just says anyone who would really use `line.dash` here would want it to be a very short dash pattern, like `'dot'`. Hardly anyone will use it, but I don't think we should prohibit it, nor should we draw different dashes in the legend vs the plot.

I've changed that mock to `dash: 'dot'` https://github.com/plotly/plotly.js/pull/2561/commits/f84dfae9d46b48fb84fd1be21c710a7dc2d346dd and I have another mock in https://github.com/plotly/plotly.js/pull/2561/commits/0b7541e34d63ce74cddb5d243a652b2df0430b70 that shows it without dashes, so users can see the normal look.

Reply:
> True, it looks weird... but it also looks weird in the plot.

Very good point. https://github.com/plotly/plotly.js/commit/f84dfae9d46b48fb84fd1be21c710a7dc2d346dd is :ok_hand: